internal/providers/terraform/azure/sentinel_data_connector_aws_cloud_trail.go

```diff
@@ -0,0 +1,28 @@
+package azure
+
+import (
+	"github.com/infracost/infracost/internal/resources/azure"
+	"github.com/infracost/infracost/internal/schema"
+)
+
+func getSentinelDataConnectorAwsCloudTrailRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_sentinel_data_connector_aws_cloud_trail",
+		RFunc: newSentinelDataConnectorAwsCloudTrail,
+		ReferenceAttributes: []string{
+			"resource_group_name",
+			"log_analytics_workspace_id",
+		},
+	}
+}
+
+func newSentinelDataConnectorAwsCloudTrail(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	region := lookupRegion(d, []string{"resource_group_name", "log_analytics_workspace_id"})
+	r := &azure.SentinelDataConnectorAwsCloudTrail{
+		Address: d.Address,
+		Region:  region,
+	}
+	r.PopulateUsage(u)
+
+	return r.BuildResource()
+}
```

Comment:
It seems like there is nothing extra in the resource structs apart from `IsSkipped` and `NoPrice` attributes. We can simplify things by defining the resources right here. For example:
```go
func getSentinelDataConnectorAwsCloudTrailRegistryItem() *schema.RegistryItem {
	return &schema.RegistryItem{
		Name:  "azurerm_sentinel_data_connector_aws_cloud_trail",
		RFunc: newSentinelDataConnectorAwsCloudTrail,
		ReferenceAttributes: []string{
			"log_analytics_workspace_id",
		},
	}
}

func newSentinelDataConnectorAwsCloudTrail(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
	return &schema.Resource{
		Name:        d.Address,
		IsSkipped:   true,
		NoPrice:     true,
		UsageSchema: []*schema.UsageItem{},
	}
}
```

With this, we don't need a bunch of files, and we can also remove `resource_group_name` reference attribute and `region` as they are not needed. This applies for all the other resources.